### DISCUSSION TITLE
Account linking with different email addresses

### DISCUSSION THREAD

**Discussion by anthamithya** [nextauthjs/next-auth#5677] [Help]
Hello NextAuth People,

I have added Email provider and Google oAuth sign in. Enabled the JWT for tokens. (not db sessions).
Logged in using Email (say, abc@xyz.com), then navigated to /api/auth/signin then logged in using Google (def@gmail.com). However, i don't see them as linked. 
My expectation is to see the same "sub" in token for both login sessions. The "sub" (in this case user id), is used to co-relate in other back-end services. 

Is this an expected behaviour? Do i need to enable dbsessions to get the same sub? Please clarify.
Thanks.

I am following the doc here at [Faq](https://next-auth.js.org/faq), saying "If the user first signs in using Email and then tries to sign in again using an OAuth provider, NextAuth.js default behavior is to allow account linking even if the OAuth account's email address does not match the previous email address of the user."

**Comment by anthamithya** [comment]
If the i login already with google then try to login with email and link the account, seems it doesn't work, as the user already created.

**Comment by anthamithya** [comment]
Another case where i feel the behaviour is surprising is follows,
1. Login with Email link abc@xyz.com
2. Navigate to /api/auth/signin and login using Google def@gmail.com
3. Accounts are linked. i.e Next time i login using Google def@gmail.com, the email minted in the token as abc@xyz.com (because of linking). And the 'sub' is same as abc@xyz.com
4. Logout
5. Login Using Email, with email as def@gmail.com, then a separate user is created. 

Is this expected/ does it pose any maintenance issues?

**Comment by nbouvrette** [comment]
Thanks for opening this discussion @anthamithya - I was looking for information about account linking and I also have another [thread](https://github.com/nextauthjs/next-auth/discussions/6162) with little activity.

I didn't know that when you try to sign-up while logged in, it will link accounts.

I tried on my end and it actually works - I suspect you have something missing on your end. Did you set `allowDangerousEmailAccountLinking: true` on your providers?
